src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/barcode/BarcodeService.kt

```diff
@@ -7,11 +7,13 @@ class BarcodeService(
   private val barcodeRepository: BarcodeRepository,
   private val barcodeEventService: BarcodeEventService,
   private val barcodeGeneratorService: BarcodeGeneratorService,
+  private val barcodeRecipientService: BarcodeRecipientService,
 ) {
 
-  fun createBarcode(userId: String): String =
+  fun createBarcode(userId: String, createBarcodeRequest: CreateBarcodeRequest?): String =
```

Comment:
I think you mentioned this previously - but do we need a transaction here and a test for if the transaction is rolled back?